### PR TITLE
Automatically publish to store

### PR DESCRIPTION
**Pull request by artivis**
Signed-off-by: artivis <jeremie.deray@canonical.com>

### CODE REVIEW

File: .github/workflows/snap.yaml [hunk 1]
```diff
@@ -28,8 +30,22 @@ jobs:
...
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: gazebo-snap
+        path: .
+    - uses: snapcore/action-publish@v1
+      with:
+        store_login: ${{ secrets.STORE_LOGIN }}
+        snap: ${{ needs.build.outputs.snap-file }}
+        release: edge
```
Comment (Guillaumebeuzeboc) [inline .github/workflows/snap.yaml:52]: Shouldn't we release on beta?

File: .github/workflows/snap.yaml [hunk 2]
```diff
@@ -28,8 +30,22 @@ jobs:
...
-    #     store_login: ${{ secrets.STORE_LOGIN }}
-    #     snap: ${{ steps.build-snap.outputs.snap }}
-    #     release: edge
+    - uses: actions/upload-artifact@v3
+      with:
+        name: gazebo-snap
+        path: ${{ steps.build-snap.outputs.snap }}
+
+  publish:
+    if: github.event_name == 'push'
```
Comment (Guillaumebeuzeboc) [inline .github/workflows/snap.yaml:39]: is a `merge` going to be considered as a push?

File: .github/workflows/snap.yaml [hunk 3]
```diff
@@ -28,8 +30,22 @@ jobs:
...
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: gazebo-snap
+        path: .
+    - uses: snapcore/action-publish@v1
+      with:
+        store_login: ${{ secrets.STORE_LOGIN }}
```
Comment (Guillaumebeuzeboc) [inline .github/workflows/snap.yaml]: This is deprecated.
Rather use:
```
    env:
      SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
```

**Review by Guillaumebeuzeboc — CHANGES_REQUESTED** [review]

File: .github/workflows/snap.yaml (hunk 2 above)
Comment (artivis) [reply .github/workflows/snap.yaml:39]: A merge is it's own event and thus is not going to be considered as a push. See e.g. the [available events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#available-events).
This is validated in the workflow run for this very pr. Only the 'build' job was run.

File: .github/workflows/snap.yaml (hunk 1 above)
Comment (artivis) [reply .github/workflows/snap.yaml:52]: Given that we're still running a beta phase at the moment, beta is therefore the main track. I thus prefer to push on edge, test and manually publish edge -> beta.

File: .github/workflows/snap.yaml (hunk 2 above)
Comment (artivis) [reply .github/workflows/snap.yaml:39]: On second though, I may have misunderstood the question. A merge request is not going to trigger the job but pushing a merge commit will.

File: .github/workflows/snap.yaml (hunk 3 above)
Comment (artivis) [reply .github/workflows/snap.yaml]: Good catch :+1:

File: .github/workflows/snap.yaml (hunk 2 above)
Comment (Guillaumebeuzeboc) [reply .github/workflows/snap.yaml:39]: So for example, only a push on citadel will trigger the publish right? but usually on this branch we don't push we merge

Comment (artivis) [reply .github/workflows/snap.yaml:39]: Either the citadel or fortress branches as defined at the top of the file.
My understanding is that merging a pr creates a merge commit which is then pushed to the branch. Thus triggering the job.

Comment (Guillaumebeuzeboc) [reply .github/workflows/snap.yaml:39]: Ok then all good

**Review by Guillaumebeuzeboc — APPROVED** [review]

**Pull request merged**